electron/main/ipcEvents.ts

```diff
@@ -69,28 +70,60 @@ export default class IpcEvents {
 
     private vanish() {
         ipcMain.on(IpcEvent.Vanish, () => {
-            log.info('Vanishing overlay');
-
-            const windowBounds = this.overlay?.getBounds() as Rectangle;
-
-            log.info('Saved window state', windowBounds);
-
-            const theme = this.store.get('savedWindowState').theme;
-
-            const data: WindowState = {
-                x: windowBounds.x,
-                y: windowBounds.y,
-                width: windowBounds.width,
-                height: windowBounds.height,
-                isClickThrough: true,
-                isTransparent: true,
-                theme,
-            };
-
-            this.store.set('savedWindowState', data);
+            if (!this.store.get('settings').isOpen && this.store.get('savedWindowState').isTransparent) {
+                // DISABLING VANISH CASE
+                // Settings are CLOSED and the Window IS transparent
+                log.info('Disabling Vanish');
+                this.store.set<typeof StoreKeys.SavedWindowState>('savedWindowState', {
+                    ...this.store.get('savedWindowState'),
+                    isClickThrough: false,
+                    isTransparent: false,
+                });
+
+                this.overlay?.setIgnoreMouseEvents(false);
+                this.overlay?.webContents.send(IpcEvent.ShowApp);
+            } else if (!this.store.get('settings').isOpen && !this.store.get('savedWindowState').isTransparent) {
+                // ENABLING VANISH CASE
+                // Settings are CLOSED and Window IS NOT transparent
+                log.info('Vanishing overlay');
+
+                const windowBounds = this.overlay?.getBounds() as Rectangle;
+
+                log.info('Saved window state', windowBounds);
+
+                const theme = this.store.get('savedWindowState').theme;
+
+                const data: WindowState = {
+                    x: windowBounds.x,
+                    y: windowBounds.y,
+                    width: windowBounds.width,
+                    height: windowBounds.height,
+                    isClickThrough: true,
+                    isTransparent: true,
+                    theme,
+                };
+
+                this.store.set('savedWindowState', data);
+
+                this.overlay?.setIgnoreMouseEvents(true);
+                this.overlay?.webContents.send(IpcEvent.Vanish);
+            } else {
```

Comment:
I noticed that the toggle also works when no channel is selected, we should catch this case as well

so only allow vanish toggle when a channel/external source is selected + the other conditions

Reply:
Yes I wanted your input for that. 
I looked into it and dropdown for example uses a way to know what page is open but I didn't understand how it knows that it is on the chatpage or any other page. 
Mostly kept it like that because of timeconstraints, since it doesn't break anything.

Reply:
the dropdown know it from the config store that is passed down into the component

```ts
const isSettingsOpen = ref(false);

if (props.store.get('settings').isOpen) {
    isSettingsOpen.value = true;
}
```

you can check it the same way probably with 

```ts
store.get('settings').isOpen
```

Reply:
looking at it makes me wonder if that needs some other refactoring too since it would touch a bunch of places that check those cases

I think for now we can keep it like it is and I'll look into refactoring this stuff